x-pack/platform/plugins/shared/actions/server/sub_action_framework/executor.ts

```diff
@@ -91,8 +93,12 @@ export const buildExecutor = <
       try {
         _subActionParams = action.schema.parse(subActionParams) as typeof subActionParams;
       } catch (reqValidationError) {
+        let errMessage = reqValidationError.message;
+        if (reqValidationError instanceof ZodError) {
+          errMessage = formatZodError(reqValidationError);
+        }
         throw createTaskRunError(
-          new Error(`Request validation failed (${reqValidationError})`),
+          new Error(`Request validation failed (${errMessage})`),
```

Comment:
This is guaranteed to run before the actual execution, right? I was thinking about the user error marking that might break when overwriting the thrown errors, but I'm pretty sure this is ok

Reply:
yes, i believe the params validation always runs before the subaction execution.